ql/src/java/org/apache/hadoop/hive/ql/parse/LoadSemanticAnalyzer.java

```diff
@@ -244,6 +245,7 @@ public void analyzeInternal(ASTNode ast) throws SemanticException {
     } else {
       analyzeLoad(ast);
     }
+    queryState.setCommandType(HiveOperation.LOAD);
```

Comment:
As far as I see we obtain a `LoadSemanticAnalyzer` from https://github.com/apache/hive/blob/e980c7a5e6ba3adec473a30e3ea0a539a5eae257/ql/src/java/org/apache/hadoop/hive/ql/parse/SemanticAnalyzerFactory.java#L75
and few lines before we set the command type using https://github.com/apache/hive/blob/e980c7a5e6ba3adec473a30e3ea0a539a5eae257/ql/src/java/org/apache/hadoop/hive/ql/parse/SemanticAnalyzerFactory.java#L62
so I suppose initially the type is set correctly to `LOAD`.

Why do need to reset the type here? Who is changing it to `QUERY` and why? Was it intentional/necessary?

Reply:
Yes for certain load queries we transform to insert into queries by calling the reparseAndSuperAnalyze method at https://github.com/apache/hive/blob/master/ql/src/java/org/apache/hadoop/hive/ql/parse/LoadSemanticAnalyzer.java#L470. 
Example, one of the cases is during the load on the bucketed tables at https://github.com/apache/hive/blob/master/ql/src/java/org/apache/hadoop/hive/ql/parse/LoadSemanticAnalyzer.java#L341

Because we transform to insert query which is always tagged as QUERY type. But we may want to preserve the actual query's type in this case, as it is what is provided by the user.

Reply:
The `HiveOperation` primarily purpose is to ensure that authorization and transaction information is passed and used correctly during the compilation and execution of Hive queries. `HiveOperation` is not meant for end-user consumption so display purposes are a secondary concern.

A `LOAD` operation and a `QUERY` operation have different implications in terms of authorization and transactions.
* Is it OK to change the query state at the end of the analysis phase? 
* Will compilation and execution run under the same authorization/transaction semantics?
* Do we have any kind of authorization/transaction tests that ensure that things work correctly for `LOAD` statements after this change?